### ISSUE TITLE
list index out of range when switching to guake

### ISSUE THREAD

**Issue by rubo77** [rubo77/set_resolution.py#4]
When I switch to `guake` by pressing F1the program crashes with:

```
line 51, in <module>
    frontmost_pid = frontmost_pid[0]
IndexError: list index out of range
```

guake uses always on top, could that be the reason somehow?

**Comment by rubo77** [comment]
I get the same error when switching back from `freemind` (java) to the standard Ubuntu `Terminal`

**Comment by Jacob-Vlijm** [comment]
I will install guake to see what happens, but what did the print line output?

**Comment by rubo77** [comment]
an empty array []

**Comment by Jacob-Vlijm** [comment]
AHA, that is useful information.

**Comment by rubo77** [comment]
Yes, strange, usually, when the switch works, 

```
        print(frontmost_pid)
```

 prints out an array like

```
['6959']
['7098']
```

but before it crashes just an empty array

**Comment by Jacob-Vlijm** [comment]
@rubo77 I wish it would happen here, I installed guake, but no matter what, it keeps running when switching from / to guake (f12). what does your resolution list look like (not sure if It matters, but I might jump into an idea). Another question: does it always happen or occasionally?
It is very well possible to rewrite a section (which I intended anyway) to catch exceptions, as a workaround, but I'd still like to understand what happens.

**Comment by rubo77** [comment]
My list file's just this: https://github.com/rubo77/set_resolution.py/blob/master/set_resolution.conf

**Comment by rubo77** [comment]
It happens really often.

maybe it has to do with the scaling of 2?

I have a resolusion of 3200x1800 and everything scaled by 2

**Comment by Jacob-Vlijm** [comment]
@rubo77  i will probably rewrite the script, test section by section for a considerate amount of time. `wmctrl` tends to produce exceptions when used repeatedly in a script, which i need to "except". I need to know however exactly where it occurs. It will probably take a few days to a week.

**Comment by rubo77** [comment]
cool, great work though!

If you need more info, I can test for you.

**Comment by Jacob-Vlijm** [comment]
I will probably ask you to run some tests :)

**Comment by Jacob-Vlijm** [comment]
Hi Ruben, I rewrote the whole story. The main goal of this version is that it should not crash. In the worst case, it will not switch incidentally, but I know how to fix that. Could you run it for a while? Would be great. https://github.com/rubo77/set_resolution.py/blob/master/set_resolution.py

**Comment by rubo77** [comment]
I tried your new version and it works with `gedit`, but it doesn’t seem to work at all with freemind, (which is java). I started the tool with:

```
 python ./set_resolution.py 
```

And when I open the Ubuntu launcher with the Super Key, I reproduceable get the error

```

[['java', '1920x1080'], ['gedit', '1920x1080']]
Traceback (most recent call last):
  File "./set_resolution.py", line 60, in <module>
    trigger2 = matching()
  File "./set_resolution.py", line 50, in matching
    matches = [item[0] for item in running if item[1] == frontmost[0].split()[2]]
IndexError: list index out of range
```

**Comment by Jacob-Vlijm** [comment]
???? Aaarrgrghh, That surprises me, I'll start testing with freemind, will rollback on AU.

**Comment by Jacob-Vlijm** [comment]
wait, you are running freemind, but what is its process named, and what is its execuable named?
Never mind, I will install it.
I would just run the script by its (absolute) path by the way.

**Comment by rubo77** [comment]
There was still an error that orccurred this time, i fixed it: https://github.com/rubo77/set_resolution.py/commit/58ce4d192cd1f588769b183b7d6a1cbba8ebf633
- And now it works with freemind again (execuable name just "java")
- No difference in using absolute path or `./`
- But still the crash, when I switch to guake with <key>F1</key>

**Comment by Jacob-Vlijm** [comment]
Hi @rubo77 Your fix is not how it is intended!! The function should return`None` in case  `len(matches) == 0`
I did make an indent mistake however, updated the script, I get no exception now, switching wildly for some time.
 F1? you mean F12? I have no Key F1 in my guake?

**Comment by rubo77** [comment]
I meant the guake function to drop down in front of everything

**Comment by rubo77** [comment]
still the same: 

It works with some windows, but when i open guake:

```
$ python set_resolution.py 
('gedit', 'xrandr -s 1920x1080')
(None, 'xrandr -s 3200x1800')
Traceback (most recent call last):
  File "set_resolution.py", line 59, in <module>
    trigger2 = matching()
  File "set_resolution.py", line 49, in matching
    matches = [item[0] for item in running if item[1] == frontmost[0].split()[2]]
IndexError: list index out of range
```

**Comment by christopherlovell** [comment]
#9 should fix this as it now loops through all associated PID values for a given window

**Comment by rubo77** [comment]
fixed with #9
